From 8b78d792ca7f7ba4e105225c3917c0a467a2fadb Mon Sep 17 00:00:00 2001
From: Keith Holman <Keith.Holman@windriver.com>
Date: Thu, 22 May 2014 12:08:13 -0400
Subject: [PATCH] barbican: increase buffer-size to support PKI tokens

With Keystone using larger PKI tokens (versus the previous UUID
tokens) causes Barbican to fail when using authentication, as
the generated packets are too large.  Increasing the maximum
buffer size uWSGI uses fixes this problem.  This fix allows the
packet sized to be specified in the build recipe file.

Signed-off-by: Keith Holman <Keith.Holman@windriver.com>
---
 etc/barbican/vassals/barbican-admin.ini | 1 +
 etc/barbican/vassals/barbican-api.ini   | 1 +
 2 files changed, 2 insertions(+)

diff --git a/etc/barbican/vassals/barbican-admin.ini b/etc/barbican/vassals/barbican-admin.ini
index e270b13..3b58430 100644
--- a/etc/barbican/vassals/barbican-admin.ini
+++ b/etc/barbican/vassals/barbican-admin.ini
@@ -8,3 +8,4 @@ no-default-app = true
 memory-report = true
 plugins = python
 paste = config:/etc/barbican/barbican-admin-paste.ini
+buffer-size = %BARBICAN_MAX_PACKET_SIZE%
diff --git a/etc/barbican/vassals/barbican-api.ini b/etc/barbican/vassals/barbican-api.ini
index 8f6f504..82de732 100644
--- a/etc/barbican/vassals/barbican-api.ini
+++ b/etc/barbican/vassals/barbican-api.ini
@@ -8,3 +8,4 @@ no-default-app = true
 memory-report = true
 plugins = python
 paste = config:/etc/barbican/barbican-api-paste.ini
+buffer-size = %BARBICAN_MAX_PACKET_SIZE%
-- 
1.9.0

